src/store.ts

```diff
@@ -89,28 +95,35 @@ class CookieStore {
     const originCookies =
       this.store.get(requestUrl.origin) || new Map<string, Cookie>()
 
-    switch (request.credentials) {
-      case 'include': {
-        // Support running this method in Node.js.
-        if (typeof document === 'undefined') {
-          return originCookies
-        }
+    try {
+      switch (request.credentials) {
+        case 'include': {
+          // Support running this method in Node.js.
+          if (typeof document === 'undefined') {
+            return originCookies
+          }
 
-        const documentCookies = parseCookie(document.cookie)
+          const documentCookies = parseCookie(document.cookie)
 
-        documentCookies.forEach((cookie) => {
-          originCookies.set(cookie.name, cookie)
-        })
+          documentCookies.forEach((cookie) => {
+            originCookies.set(cookie.name, cookie)
+          })
 
-        return originCookies
-      }
+          return originCookies
+        }
 
-      case 'same-origin': {
-        return originCookies
-      }
+        case 'same-origin': {
+          return originCookies
+        }
 
-      default:
-        return new Map()
+        default:
+          return new Map()
+      }
+    } catch {
+      // Ignore errors thrown by `request.credentials` access, such as those in cloudflare workers
+      // We can't just check that `credentials` is in request, because it is in cloudflare workers test environment
+      // however access throws an error
+      return originCookies
```

Comment:
It'd be great to find a more deterministic way of detecting those Cloudflare errors and doing an early return instead of wrapping the logic in a `try/catch` block. 

@mattcosta7, can please include what exact error is being thrown here and when?

Reply:
the error is `xxxxxx is not implemented` 

https://github.com/cloudflare/miniflare/blob/18379377aeb276b246814870422e7d57f743ac95/packages/core/src/standards/http.ts#L516-L531

https://github.com/cloudflare/miniflare/blob/18379377aeb276b246814870422e7d57f743ac95/packages/core/src/standards/http.ts#L721-L727

since these are defined as getters, property access throws directly, and `Reflect.has` should succeed, similar to `'credentials' in request` type checks.   I'm not sure there's a better way to check that access doesn't throw in these cases?   it does seem odd to me that they'd choose to throw for these properties instead of just not implement them at all, but I imagine this is because types won't support their usage easily by default

Reply:
Then I'm all for using `Reflect.has()` to determine if the property exists instead of `try/catch`. Would that achieve the goal here? 

```js
isCredentialsImplemented = Reflect.has(request, 'credentials')

if (!isCredentialsImplemented) {
  return originalCookies
}
```

Reply:
> Then I'm all for using `Reflect.has()` to determine if the property exists instead of `try/catch`. Would that achieve the goal here?
> 
> ```js
> isCredentialsImplemented = Reflect.has(request, 'credentials')
> 
> // A nice comment here describing why we check for this. 
> if (!isCredentialsImplemented) {
>   return originalCookies
> }
> ```
 
but `credentials` _is implemented_ and getters respond to Reflect.has

```ts
class MyRequest {
   get credentials() {}
}

Reflect.has(new MyRequest(), 'credentials') 
=> true
```

I don't believe there's any way to test for "this getter is implemented _and doesn't throw_" without a try/catch

Reply:
When I said `Reflect.has` should succeed, I mean it will be truthy even though we'd not want it to be. 

Reflect.has in this case is just about the same as an `in` check, and the Request/Response objects in miniflare _have the properties defined_ they just throw on access

Reply:
Got it. Thanks for the explanation. 

What do you think if we move the "is implemented" check to its own utility? I really think we should check it once and do an early return.

```js
export function isImplemented(target, propertyName) {
  try {
    target[propertyName]
    return true
  } catch (_error) {
    return false
  }
}
```

Reply:
yep, definitely can do that too!

Reply:
updated in https://github.com/mswjs/cookies/pull/27/commits/758c5eb0aa322a0438a4580b07bd0f02c9a1ef0a